spec/services/role_checker_spec.rb

```diff
@@ -0,0 +1,78 @@
+RSpec.describe RoleChecker, type: :service do
+  let(:records){ [] }
+  let(:user) { create(:user, roles: {'123': ['tester'], '456': ['expert']}) }
+  let(:checker){ described_class.new user, records }
+  let(:resource_double) { double(id: '123')}
+  let(:another_resource_double) { double(id: '456')}
+  let(:one_more) { double(id: '666')}
+
+  describe '#user_role_ids' do
```

Comment:
this is really a private implementation method, you shouldn't be testing this implementation detail and it shouldn't be in the public API (it's only in use in the role Checker)

Reply:
I always err on the side of too many specs, so good catch